examples/seismic/source.py

```diff
@@ -1,60 +1,116 @@
+from scipy import signal
+
 from devito import Dimension
 from devito.function import SparseTimeFunction
 from devito.logger import error
 
+from cached_property import cached_property
+
+from copy import copy
 import numpy as np
 try:
     import matplotlib.pyplot as plt
 except:
     plt = None
 
-__all__ = ['PointSource', 'Receiver', 'Shot', 'RickerSource', 'GaborSource']
+__all__ = ['PointSource', 'Receiver', 'Shot', 'WaveletSource',
+           'RickerSource', 'GaborSource', 'TimeAxis']
+
+
+class TimeAxis:
+    """ Data object to store the time axis.
+
+    :param start: Start of time axis.
+    :param step: Time step. This should be set to None if stop is set.
+    :param: num: Number of times steps.
+    :stop: (Optional) End time.
+    """
+    def __init__(self, start, step, num, stop=None):
+        if stop is not None:
+            if step is not None:
+                raise ValueError("step should be set to None if stop is set")
+            step = (stop-start)/num
+
+        self.start = start
+        self.stop = start + step*num
+        self.step = step
+        self.num = num
+
+    def __str__(self):
+        return "TimeAxis: start=%g, stop=%g, step=%g, num=%g" % \
+               (self.start, self.stop, self.step, self.num)
+
+    @cached_property
+    def time_values(self):
+        return np.linspace(self.start, self.stop, self.num, endpoint=False)
```

Comment:
I still would put `endpoint=True` to be consistent with the previous computations of `start/end/step`